src/widgets/lv_img.c

```diff
@@ -67,6 +67,11 @@ lv_obj_t * lv_img_create(lv_obj_t * parent)
  *====================*/
 
 void lv_img_set_src(lv_obj_t * obj, const void * src)
+{
+    lv_img_set_src_ex(obj, src, NULL);
+}
+
+void lv_img_set_src_ex(lv_obj_t * obj, const void * src, void * dec_ctx)
```

Comment:
I'm not sure about this. Ideally the lottie animation should use the width/height of the image object. Is there a way to pass it to the decoder from here?

Reply:
If you use NULL (or just `set_src`), it'll be created with the picture's internal size. The image size will be deduced from there.
If you resize the object, does it invalidate the cache ? I'm not sure. 
If it does, then the next time decoder_open will be called, it'll rebuild the buffer.
I guess we need a way to tell the vector engine to reallocate it. Let me think about it.

Reply:
Could link me code section where the image object's size is passed to the decoder?

We have `LV_EVENT_SIZE_CHANGED` so we can invalidate the cache there. Maybe we can a flag in `dec_ctx` to tell whether this source is "size change sensitive" or not.

Reply:
Good idea. Adding a flag in the `img_decoder_context_t` would work for this.
Look at `lv_rlottie`'s constructor. I'm changing the `create_width` and `create_height` member here. 
I'll hack something tomorrow.

Reply:
> Look at lv_rlottie's constructor. I'm changing the create_width and create_height member here.
I'll hack something tomorrow.

How it works if you use the rlottie file with `lv_img_set_src`? Does the user need to set up a ctx too and call `lv_img_set_src_ex`?

Reply:
No, it works. 

When implementing, I've tested this first (before increasing the difficulty :-). The lottie file contains a width and height. If no context is provided (when you call `lv_img_set_src`) then this file's size and the first frame of the animation are used.

So it means that you can use a rlottie file in any official image now (like in imgbtn, btnmatrix, background image, etc...) and it'll work. 
Only when you need to change the frame to display, or the size to render, you'll need to deal with the decoder context.
So the `lv_rlottie` file is just a wrapper to deal with animation logic only, so it's easier to use than having to change the decoder context's current_frame and invalidating the image by yourself. In fact, there is no such thing as rlottie specific in `lv_rlottie` anymore (except for destruction).

The `lv_animbtn` is a wrapper over `lv_rlottie` to deal with mapping button's state with animation parts.
Technically, once this PR is merged, it'll be very easy to use the anim button with sjpg as well.

Reply:
I see. It'd great the rlottie has a "native size". 

Now it's a little bit messy how the image object's size affects the image source. In the current version the images start to tile to fill the widget size. However at least these modes should work:
- do nothing (leave in top left)
- center
- zoom to fill object size
- tile

So normally setting a larger widget size tiles the image but we want rlottie to fit the new size. Its clear that there is a need for adjusting the methods above.

Anyway, `LV_EVENT_SIZE_CHANGED`  might be used to change the rlottie animation size to the new widget size.

BTW, do have any ideas how to handle `lv_img_set_zoom/angle()` on rlottie?

>  it'll be very easy to use the anim button with sjpg as well.

sjpg is not animated, "just" a line-by-line JPEG decoder.

Reply:
Interesting. I never used sjpg before and reading the code, I thought the frame_index was actual frames. I don't see the point of this format compared to plain old jpeg that's also doing 8x8 blocs ???

> but we want rlottie to fit the new size.

Do we ?
Currently, it'll follow what the other image format are doing, since there's no size feedback.

Doing a digital zoom on a vector format seems dumb, so yes, we might want to be smarter here (same for rotating, we can compute it in the rlottie's matrix). However, I see an issue here with clipping or changing the rendering area dimensions (and it'll require a patch to rlottie code too to transmit the rotation angle & center). It's not unfeasible but it's a lot more work.

So, for now, let's stay simple here and only handle zoom nicely. When zooming, we can check the lv_img_decoder_context's flag, find out it's a vector based format, adjust the create_width/create_height accordingly, invalidate the cache and re-open the decoder. It'll rebuild the rendering buffer for the new size and it'll work.

Reply:
There's no special treatment for rotate yet. But for zoom, it's supposed to be done now for vector format.
Please check.